mono/mini/aot-compiler.c

```diff
@@ -9072,7 +9072,12 @@ append_mangled_wrapper (GString *s, MonoMethod *method)
 	case MONO_WRAPPER_MANAGED_TO_NATIVE: {
 		append_mangled_wrapper_subtype (s, info->subtype);
 		if (info->subtype == WRAPPER_SUBTYPE_ICALL_WRAPPER) {
-			g_string_append_printf (s, "%s", method->name);
+			const char *name = method->name;
+			const char *prefix = "__icall_wrapper_";
+			if (strstr (name, prefix) == name)
```

Comment:
`strstr` is sloppy and slow here. You basically want a length check and then memcmp, except for maybe a varying number of leading underscores.